arangod/RocksDBEngine/RocksDBTransactionState.cpp

```diff
@@ -141,26 +136,25 @@ Result RocksDBTransactionState::beginTransaction(transaction::Hints hints) {
   _counterGuard = mgr->registerTransaction(id(), isReadOnlyTransaction(),
                                            isFollowerTransaction());
 
-  TRI_ASSERT(_cacheTx == nullptr);
+  TRI_ASSERT(_cacheTx.term == cache::Transaction::kInvalidTerm);
 
   // start cache transaction
   auto* manager =
       vocbase().server().getFeature<CacheManagerFeature>().manager();
   if (manager != nullptr) {
-    _cacheTx = manager->beginTransaction(isReadOnlyTransaction());
+    manager->beginTransaction(_cacheTx, isReadOnlyTransaction());
   }
 
   return res;
 }
 
 void RocksDBTransactionState::cleanupTransaction() noexcept {
-  if (_cacheTx != nullptr) {
+  if (_cacheTx.term != cache::Transaction::kInvalidTerm) {
     // note: endTransaction() will delete _cacheTrx!
```

Comment:
This comment is now misleading and should be removed.

Reply:
Fixed